### DISCUSSION TITLE
Face detection of similar faces

### DISCUSSION THREAD

**Discussion by pabera** [immich-app/immich#14596] [Q&A]
Similar to this request #13311, I have a 2 siblings in my library. They are a couple of years apart. They looked very similar as babies of course. And Immich doesn't recognize them separately.

Immich has grouped both their faces under one, 10k+ photos. Is there a way I can get them separated? 
I started to reassign them manually (would take a bit of time), but I notice that newly added photos are still recognized wrongly.

How does it work under the hood? If I am able to reassign more photos manually, would Immich start to be able to  distinguish them at some point? I just want to avoid that I put time into this and in the end, the ML has learned the wrong thing.

Since both kids have a separate age, maybe there is a way to ask the ML to learn until a specific date (until the second kid is born)

**Comment by bquorning** [comment]
I have the exact same problem. I have two boys about 2 years apart. They are now 14 and 16, and according to Immich I have 23k pictures of the older and 1.5k pictures of the younger (so probably 10k pictures have the wrong face added). Changing each wrong face is at least 4 clicks, so I too am hoping there’s an easier way to fix this.

**Comment by skatsubo** [comment]
You can try slightly lowering "Max Recognition Distance" https://immich.app/docs/features/facial-recognition/#maximum-recognition-distance

> The default works well for most people, but it may be worth lowering it if the library has twins or otherwise very similar looking people. 

Similar discussions:
- https://www.reddit.com/r/immich/comments/192o8l6/facial_recognition_brothers/
- https://github.com/immich-app/immich/discussions/3946

**Reply by pabera** [reply]
I case rerunning the face recognition is not working (it takes too much time, time spend on naming and merging would be lost), I have come up with a small tool to fix face assignments manually and partially automatically.

* https://github.com/pabera/immich-face-fix

**Reply by ghost** [reply]
"partially automatically" how does it work?
